src/GitHub.UI/Helpers/SharedDictionaryManagerBase.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Windows;
+using System.Collections.Generic;
+
+namespace GitHub.Helpers
+{
+    public class SharedDictionaryManagerBase : ResourceDictionary
+    {
+        static IDictionary<Uri, ResourceDictionary> sharedDictionaries;
+        static IList<IDisposable> disposables;
```

Comment:
You can use a `CompositeDisposable` type here, it's designed to handle disposables in an optimal way (you can just call `Dispose()` on it and it will handle disposing everything it holds.